src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -273,13 +263,13 @@ ObjectDataHandler::write_ret ObjectDataHandler::prepare_data_reservation(
     DEBUGT("reserving: {}~{}",
            ctx.t,
            ctx.onode.get_data_hint(),
-           MAX_OBJECT_SIZE);
+           max_object_size);
     return ctx.tm.reserve_region(
       ctx.t,
       ctx.onode.get_data_hint(),
-      MAX_OBJECT_SIZE
-    ).si_then([&object_data](auto pin) {
-      ceph_assert(pin->get_length() == MAX_OBJECT_SIZE);
+      max_object_size
+    ).si_then([max_object_size=max_object_size, &object_data](auto pin) {
```

Comment:
[max_object_size, &object_data]

Reply:
max_block_size is not a local variable but a member of object data handler